Models/Connection/MySqlConnectionInfo.js

```diff
@@ -0,0 +1,200 @@
+                temp.value = fieldValue;
+                tvp.push(temp);
+              }
+            }
+            input[key] = tvp;
+          } else {
+            input[key] = value;
+          }
+        }
+      }
+      await this.createTemporaryTableAndInsertData(
+        "temp_table",
+        input.params,
+      );
+
+      const procedureResult = await this.executeProcedure(
+        this.procedure,
+        "temp_table"
+      );
+
+      result = new DataSourceCollection(procedureResult);
+      resolve(result);
+    });
+  }
+
+  /**
+   * @param {Request} request
+   * @param {CancellationToken} cancellationToken
+   * @returns {Promise<IDataSource>}
+   */
+  async getRoutingDataAsync(request, cancellationToken) {
+    const params = [];
+    for (const type in request) {
+      const group = request[type];
+      for (const name in group) {
+        if (name === "query") {
+          const query = group[name];
+          for (const key in query) {
+            const row = new MysqlRow(name, key, query[key].toString());
+            params.push(row);
+          }
+        } else {
+          const row = new MysqlRow(type, name, group[name].toString());
+          params.push(row);
+        }
+      }
+    }
+
+    const result = await this.loadDataAsync(
+      { params: params },
+      cancellationToken
+    );
+    const retVal = {};
+
+    if (result && result.items && result.items[0] && result.items[0].data) {
+      result.items[0].data.forEach((row) => {
+        if (!retVal[row.paramType]) {
+          retVal[row.paramType] = {};
+        }
+
+        retVal[row.paramType][row.paramKey] = row.paramValue;
+      });
+    }
+    return retVal;
+  }
+
+  /**
+   * @param {string} pageName
+   * @param {string} rawCommand
+   * @param {number} pageSize
+   * @param {number} domainId
+   * @param {CancellationToken} cancellationToken
+   * @returns {Promise<ILoadPageResult>}
+   */
+  async loadPageAsync(
+    pageName,
+    rawCommand,
+    pageSize,
+    domainId,
+    cancellationToken
+  ) {
+    /** @type {mySql.ConnectionPool?} */
+    this.pool.getConnection(async (err, connection) => {
+      if (err) {
+        throw err;
+      }
+
+      const rows = await this.executeProcedure(
+        connection,
+        this.settings.procedure,
+        pageName,
+        domainId,
+        pageSize,
+        rawCommand
+      );
+      if (rows.length != 1) {
+        throw new WebServerException(
+          `Call Command Expect 1 File '${pageName}' But Get ${rows.length} File(s) From '${this.settings.procedure}' Procedure`
+        );
+      }
+      return rows[0];
+    });
+  }
+  /**
+   * Executes a stored procedure with the provided parameters using a MySQL connection.
+   * @param {string} procedureName
+   * @param {Array} parameters
+   * @returns {Promise<any>}
+   * @throws {Error}
+   */
+  executeProcedure(procedureName, parameters) {
+    return new Promise((resolve, reject) => {
```

Comment:
We do not need create Promise object. use async version of mysql provider method:
```javascript
// get the client
  const mysql = require('mysql2');
  // create the pool
  const pool = mysql.createPool({host:'localhost', user: 'root', database: 'test'});
  // now get a Promise wrapped instance of that pool
  const promisePool = pool.promise();
  // query database using promises
  const [rows,fields] = await promisePool.query("SELECT 1");```